share/config.fish

```diff
@@ -154,15 +158,14 @@ end
 # Upgrade pre-existing abbreviations from the old "key=value" to the new "key value" syntax.
 # This needs to be in share/config.fish because __fish_config_interactive is called after sourcing
 # config.fish, which might contain abbr calls.
-if not set -q __fish_init_2_3_0
+if test $__fish_initialized -lt 2300 && not set -q __fish_init_2_3_0
```

Comment:
So thinking more about this I'd initialize $__fish_initialized with the highest old variable.

So you'd do something like

```fish
if not set -q __fish_initialized
    if set -q __fish_init_2_39_8
        set -U __fish_initialized 2398
    else if set -q __fish_init_2_3_0
        set -U __fish_initialized 2300
    end
end
```

It's not massively shorter, but it localizes the knowledge about the old variables - so you can just rely on $__fish_initialized.

Reply:
Done, thanks!

I also changed it now to initialize to `3100` instead of `3010`, that makes more sense.